### PR TITLE
Adding script to deploy HCO in one line

### PR DESCRIPTION
**Pull request by mmirecki**

### CODE REVIEW

File: deploy/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
```
Comment (ohadlevy) [inline deploy/deploy.sh]: how about adding ```set -e``` ? (http://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#The-Set-Builtin)

File: README.md [hunk 2]
```diff
@@ -21,9 +21,16 @@ Until we publish (and consume) the HCO and component operators through
 Marketplace|operatorhub.io, this is a means to demonstrate the HCO workflow
 without OLM.
 
+You can either run the following script:
+```bash
+deploy/deploy.sh
+```
+or execute the steps below manually.
```
Comment (dankenigsberg) [inline README.md]: I think it is important to NOT to keep the steps here. If we do, we end up needed to keep to places up to date.

File: deploy/deploy.sh (hunk 1 above)
Comment (dankenigsberg) [reply deploy/deploy.sh]: +1
ignoring errors is impolite.

File: deploy/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+kubectl create ns kubevirt-hyperconverged
+kubectl create ns node-maintenance-operator
+
+kubectl config set-context $(kubectl config current-context) --namespace=kubevirt-hyperconverged
+
```
Comment (dankenigsberg) [inline deploy/deploy.sh]: you can add the human text here as comments e.g.

`# Launch all of the CRDs.`

File: deploy/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (rthallisey) [inline deploy/deploy.sh]: `hack/deploy.sh` already covers most of this.  You just need to remove the component CR creation on lines 50, 55, and 60.

File: README.md [hunk 5]
```diff
@@ -21,36 +21,12 @@ Until we publish (and consume) the HCO and component operators through
...
-Launch all of the CRDs.
 ```bash
-kubectl create -f deploy/converged/crds/hco.crd.yaml
-kubectl create -f deploy/converged/crds/kubevirt.crd.yaml
-kubectl create -f deploy/converged/crds/cdi.crd.yaml
-kubectl create -f deploy/converged/crds/cna.crd.yaml
-kubectl create -f deploy/converged/crds/ssp.crd.yaml
-kubectl create -f deploy/converged/crds/kwebui.crd.yaml
-kubectl create -f deploy/converged/crds/nodemaintenance.crd.yaml
+deploy/deploy.sh
```
Comment (rthallisey) [inline README.md]: hack/deploy.sh

File: deploy/deploy.sh (hunk 4 above)
Comment (mmirecki) [reply deploy/deploy.sh]: Another issue when just running hack/deploy by itself (not part of a make target) is that _out is empty. The new deploy can be used on any cluster, not created by a hco make.

**Comment by mmirecki** [comment]
My intent is to have a script with which I can quickly deploy the hco operator on any env.
My current usecase is:
- deploy an env (for example on aws)
- clone this repo
- run script
With the hack/deploy.sh script, this will not work (as the _out dir will not exist). You first need to run some make targets before it will be usable.

File: deploy/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+# Create the namespaces for the HCO
+kubectl create ns kubevirt-hyperconverged
+kubectl create ns node-maintenance-operator
+
+# Switch to the HCO namespace.
+kubectl config set-context $(kubectl config current-context) --namespace=kubevirt-hyperconverged
+
+# Launch all of the CRDs.
+kubectl create -f deploy/converged/crds/hco.crd.yaml
```
[the remark this answers is not in the document]
Comment (mmirecki) [reply deploy/deploy.sh]: Good hint! Changing to links.

File: deploy/deploy.sh [hunk 7]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
```
Comment (rthallisey) [inline deploy/deploy.sh]: let's also have bash report what command is running.  `set -ex`

File: README.md (hunk 5 above)
Comment (rthallisey) [reply README.md]: $ curl https://raw.githubusercontent.com/kubevirt/hyperconverged-cluster-operator/deploy/deploy.sh | bash

File: README.md [hunk 8]
```diff
@@ -21,36 +21,12 @@ Until we publish (and consume) the HCO and component operators through
...
-Create the namespace for the HCO.
-```bash
-kubectl create ns kubevirt-hyperconverged
-```
-
-Switch to the HCO namespace.
-```bash
-kubectl config set-context $(kubectl config current-context) --namespace=kubevirt-hyperconverged
-```
+Run the following script to apply the HCO operator:
```
Comment (rthallisey) [inline README.md]: `$ curl https://raw.githubusercontent.com/kubevirt/hyperconverged-cluster-operator/deploy/deploy.sh | bash`

File: deploy/deploy.sh (hunk 7 above)
Comment (mmirecki) [reply deploy/deploy.sh]: done

File: README.md (hunk 8 above)
Comment (mmirecki) [reply README.md]: done

**Comment by rthallisey** [comment]
Thank you @mmirecki !

**Pull request merged**